Modules/Module_Processing/xASL_init_LoadDataPar.m

```diff
@@ -0,0 +1,133 @@
+% DESCRIPTION: Load data parameter file (which contains settings for running the pipeline, 
+%              so should actually be called something like xASLProcessingSettings.json
+%
+% EXAMPLE:     This is part of the initialization workflow. Check out the usage there.
+%
+% This function performs the following parts:
+% 1. Generate warning for incompatibility with old dataPar.m
+% 2. Choose the dataPar location
+% 3. Load pre-existing dataPar
+% 4. Populate dataPar with missing parameters
+%
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% REFERENCES:  n/a
+% __________________________________
+% Copyright (c) 2015-2024 ExploreASL
+    
+    
+
+    %% 1. Generate warning for incompatibility with old dataPar.m
+    [~, ~, Dext] = fileparts(x.dir.dataPar);
+    if strcmp(Dext,'.m')
+        warning('No .m file backwards compatibility starting v1.10.0...');
+    elseif strcmp(Dext,'.mat')
+        warning('No .mat file backwards compatibility starting v1.10.0...');
+    end
+
+
+    %% 2. Choose the dataPar location
+
+    bUseRoot = false;
+    bUseRawdata = false;
+    bUseDerivatives = false;
+
+    % Check dataPar inside the root folder /
+    listRoot = xASL_adm_GetFileList(x.dir.DatasetRoot, '(?i)(^dataPar.*\.json$)', 'FPList', [], 0);
+    if ~isempty(listRoot)
+        bUseRoot = true;
+    end   
+
+    % Check dataPar inside the rawdata folder /rawdata
+    listRawdata = xASL_adm_GetFileList(x.dir.RawData, '(?i)(^dataPar.*\.json$)', 'FPList', [], 0);
+    if ~isempty(listRawdata)
+        warning([xASL_num2str(length(listRawdata)) ' dataPar.json (or similar) file(s) found in ' x.dir.RawData ', will try this but this is not the appropriate location']);
+        bUseRawdata = true;
+    end
+
+    % Check dataPar inside the processing folder /derivatives/ExploreASL
+    fListLegacy = xASL_adm_GetFileList(x.dir.xASLDerivatives, '(?i)(^dataPar.*\.json$)', 'FPList', [], 0);
+    if ~isempty(fListLegacy)
+        bUseDerivatives = true;
+    end
+
+    % Choose folder & filelist
+    if bUseDerivatives && bUseRoot
+            % We prioritize existing dataPar.json in the derivatives folder
+        fprintf('%s\n', 'dataPar.json (or similar) file(s) both in /derivatives/ExploreASL and / root folder, ignoring root folder');
+        bUseRoot = false;
+    end
+    if bUseDerivatives
+        fFolder = x.dir.xASLDerivatives;
+        listDatapar = fListLegacy;
+        fprintf('%s\n', ['dataPar.json found in ' fFolder]);
+    elseif bUseRoot
+        fFolder = x.dir.DatasetRoot;
+        listDatapar = listRoot;
+        fprintf('%s\n', ['dataPar.json found in ' fFolder]);
+    elseif bUseRawdata
+        fFolder = x.dir.RawData;
+        listDatapar = listRawdata;
+        fprintf('%s\n', ['dataPar.json found in ' fFolder]);
+    else
+        listDatapar = {};
```

Comment:
In case no dataPar is discovered the it still tries to read it on line 97 and then crashes.

Reply:
Oops :)

Fixed in 2105779ffe8d77ebeef8f57150c086e5cbedfb1a